pkg/policies/binary/binary.go

```diff
@@ -0,0 +1,212 @@
+// scorecard.
+package binary
+
+import (
+	"context"
+	"fmt"
+	"path"
+
+	"github.com/ossf/allstar/pkg/config"
+	"github.com/ossf/allstar/pkg/config/operator"
+	"github.com/ossf/allstar/pkg/policydef"
+
+	"github.com/google/go-github/v32/github"
+	"github.com/ossf/scorecard/checker"
+	"github.com/ossf/scorecard/checks"
+	"github.com/ossf/scorecard/clients/githubrepo"
+	"github.com/rs/zerolog/log"
+)
+
+const configFile = "binary_artifacts.yaml"
+const polName = "Binary Artifacts"
+
+// OrgConfig is the org-level config definition for this policy.
+type OrgConfig struct {
+	// OptConfig is the standard org-level opt in/out config, RepoOverride applies to all
+	// config.
+	OptConfig config.OrgOptConfig `yaml:"optConfig"`
+
+	// Action defines which action to take, default log, other: issue...
+	Action string `yaml:"action"`
+}
+
+// RepoConfig is the repo-level config for this policy.
+type RepoConfig struct {
+	// OptConfig is the standard repo-level opt in/out config.
+	OptConfig config.RepoOptConfig `yaml:"optConfig"`
+
+	// Action overrides the same setting in org-level, only if present.
+	Action *string `yaml:"action"`
+}
+
+type mergedConfig struct {
+	Action string
+}
+
+type details struct {
+	Messages []checker.CheckDetail
+}
+
+var configFetchConfig func(context.Context, *github.Client, string, string, string, interface{}) error
+
+func init() {
+	configFetchConfig = config.FetchConfig
+}
+
+// Binary is the Binary Artifacts policy object, implements policydef.Policy.
+type Binary bool
+
+// NewBinary returns a new Binary Artifacts policy.
+func NewBinary() policydef.Policy {
+	var b Binary
+	return b
+}
+
+// Name returns the name of this policy, implementing policydef.Policy.Name()
+func (b Binary) Name() string {
+	return polName
+}
+
+type logger struct {
+	Messages2 []checker.CheckDetail
+}
+
+func (l *logger) Info(desc string, args ...interface{}) {
+	cd := checker.CheckDetail{Type: checker.DetailInfo, Msg: fmt.Sprintf(desc, args...)}
+	l.Messages2 = append(l.Messages2, cd)
+}
+
+func (l *logger) Warn(desc string, args ...interface{}) {
+	cd := checker.CheckDetail{Type: checker.DetailWarn, Msg: fmt.Sprintf(desc, args...)}
+	l.Messages2 = append(l.Messages2, cd)
+}
+
+func (l *logger) Debug(desc string, args ...interface{}) {
+	cd := checker.CheckDetail{Type: checker.DetailDebug, Msg: fmt.Sprintf(desc, args...)}
+	l.Messages2 = append(l.Messages2, cd)
+}
+
+// Check performs the polcy check for this policy based on the
+// configuration stored in the org/repo, implementing policydef.Policy.Check()
+func (b Binary) Check(ctx context.Context, c *github.Client, owner,
+	repo string) (*policydef.Result, error) {
+	oc, rc := getConfig(ctx, c, owner, repo)
+	enabled := config.IsEnabled(oc.OptConfig, rc.OptConfig, repo)
+	log.Info().
+		Str("org", owner).
+		Str("repo", repo).
+		Str("area", polName).
+		Bool("enabled", enabled).
+		Msg("Check repo enabled")
+
+	repoClient := githubrepo.CreateGithubRepoClient(ctx, c)
```

Comment:
Use `RunScorecards` fn directly?

https://github.com/ossf/scorecard/blob/main/pkg/scorecard.go#L72

Reply:
Looking at `RunScorecards` I think it includes some code that I don't necessarily want to run in Allstar. Lets continue the discussion of what would be a nice clean way to interface and develop a plan.